### PR TITLE
[recnet-api] arXiv integration

### PR DESCRIPTION
**Pull request by joannechen1223**
## Description

This PR is mainly working on the feature of arXiv integration, which allows users to paste a random arXiv link and get the metadata directly so they don't need to fill in by themselves. To implement this feature with the scalability of different academic sites(e.g. IEEE, ACM) in the future, this PR introduces the concept of `DigitalLibrary` and mainly uses regular expressions to match the given link.

For more information, please refer to the [design document](https://www.notion.so/Digital-Libraries-Integration-arXiv-26e1673576a84541bb4d789fea2ff518).

This PR also initiates the unit tests of `recnet-api`. I changed some configurations to fix some ci unit tests issues.

## Related Issue

- https://github.com/lil-lab/recnet/issues/256
- https://github.com/lil-lab/recnet/issues/254

### How to test locally?

1. Set up your local DB and run the following SQL command to create the arXiv digital library service manually.
```sql
INSERT INTO recnet."DigitalLibrary" ("name", regex, "isVerified", "rank")
VALUES (
  'arXiv', 
  '{"https?:\/\/arxiv\.org\/(abs|pdf)\/(?<id>[0-9.]+(v[0-9]+)?)"}',
  true,
  1
);
```
2. Hit `GET /article` API and put arXiv URL in link query param

- Case 1: Article Link is an arXiv unified link and exists in DB => should return DB value
- Case 2: Article Link is an arXiv unified link and does NOT exist in DB => should fetch arXiv API to get the metadata
  - https://arxiv.org/abs/2409.13688
  - https://arxiv.org/abs/2409.13688v1 (with version)
- Case 3: Article Link is an arXiv pdf link and its unified link exists in DB => should return DB value
  - https://arxiv.org/pdf/2409.13688
  - https://arxiv.org/pdf/2409.13688.pdf
  - https://arxiv.org/pdf/2409.13649v1
  - https://arxiv.org/pdf/2409.13649v1.pdf
- Case 4: Article Link is NOT an arXiv link and exists in DB => should return DB value
- Case 5: Article Link is NOT an arXiv lin and does NOT exist in DB => should return null

## TODO

- [ ] Paste the testing link
- [x] Clear `console.log` or `console.error` for debug usage
- [x] Unit tests
- [x] Revise Notion document
- [ ] Insert to arXiv data row to db manually

### CODE REVIEW

File: apps/recnet-api/src/modules/article/article.service.ts [hunk 1]
```diff
@@ -1,27 +1,71 @@
...
+  ): CreateArticleInput {
     return {
-      article: article,
+      link,
+      doi: null,
+      title: metadata.title || "",
+      author: metadata.author || "",
+      year: metadata.year || 0,
+      month: metadata.month || null,
+      isVerified: metadata.isVerified || false,
```
[the remark this answers is not in the document]
Comment (swh00tw) [reply apps/recnet-api/src/modules/article/article.service.ts:68]: And remember to change some test cases after you add some logics to discard "invalid metadata".

**Review by swh00tw — CHANGES_REQUESTED** [review]
Looks good, only few things need to change

File: .github/workflows/ci.yml
Comment (swh00tw) [inline .github/workflows/ci.yml:1]: Forget to mention that you can actually write "dependsOn" in `recnet-api`'s `project.json` to run some targets before running target `test`. See example in `recnet-jwt/project.json`

File: apps/recnet-api/jest.setup.ts
[the remark this answers is not in the document]
Comment (joannechen1223) [reply apps/recnet-api/jest.setup.ts:1]: fixed

File: apps/recnet-api/src/modules/article/article.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (joannechen1223) [reply apps/recnet-api/src/modules/article/article.service.ts]: fixed

File: .github/workflows/ci.yml
Comment (joannechen1223) [reply .github/workflows/ci.yml:1]: fixed

**Review by swh00tw — APPROVED** [review]
Great works!

**Pull request merged**